adding boolean functions for git status

tarcinil opened the pull request:
fixes #346

tarcinil commented:
The test that failed is outside the scope of the change that I made.  I will look into forcing a Travis run to see if it can be replicated twice.

perlun commented on lib/git/status.rb:
```diff
@@ -9,19 +9,35 @@ def initialize(base)
+    end
+
+    def untracked?(file)
+      untracked.member?(file)
```
For the new methods added, would you mind adding YARD-style comments that describes briefly:

- What they do
- What the expected types of the input and return values are

The new methods added in this PR are:

- changed?
- added?
- deleted?
- untracked?

Something like this will do:

```ruby
#
# Determines whether the given file has been added to the repository
#
# @param file [String] The name of the file.
# @return [Boolean]
```

This way, it is clearer for readers of the code what the intent of these (public API methods) is.

tarcinil commented:
@perlun 

I took your comments to heart and while I was adding the documentation, I fixed the class so that it would pass rubocop.

```bash
vburton@vburton-VirtualBox:~/Code/ruby-git$ rubocop lib/git/status.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

The construct_status was broken up to lower method complexity while creating methods to be responsible for untracked, modified, and added
